pkg/tsdb/cloudwatch/log_query.go

```diff
@@ -27,9 +26,9 @@ func logsResultsToDataframes(response *cloudwatchlogs.GetQueryResultsOutput) (*d
 
 				// Check if field is time field
 				if _, err := time.Parse(cloudWatchTSFormat, *resultField.Value); err == nil {
-					fieldValues[*resultField.Field] = make([]*time.Time, rowCount)
+					fieldValues[*resultField.Field] = make([]*time.Time, 0)
```

Comment:
not sure if this is going to be less efficient or not, but basically due to empty rows we don't know the length beforehand. Could do one more pass on Results to filter them out but that seems less efficient to me on first glance.

Reply:
@aocenas Actually, to be efficient about memory allocation, you can use array slice capacity instead of length: 

```suggestion
					fieldValues[*resultField.Field] = make([]*time.Time, 0, rowCount)
```